### ISSUE TITLE
Email begins with # does not match

### ISSUE THREAD

**Issue by Mehmet-Erkan** [dotnet/docs#5305]
#myName@email.com will not match from the regex above. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 202e3f9f-a12a-53de-9069-fd469e3dbebc
* Version Independent ID: fd3699bb-4881-b6b4-1275-43eed3d5f626
* Content: [How to: Verify that Strings Are in Valid Email Format](https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format)
* Content Source: [docs/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format.md](https://github.com/dotnet/docs/blob/master/docs/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format.md)
* Service: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

**Comment by dasMulli** [comment]
apart from that, is email really a good example for RegEx usage?

IMHO, docs could (should?) point to `System.Net.Mail.MailAddress` for email addresses, even though there is no `TryParse` method (https://github.com/dotnet/corefx/issues/25295).

**Comment by mairaw** [comment]
@terrajobst @weshaggard should we update that topic to use the API instead of regexes? That topic has a lot of page views but low CSAT, so it would be good to address this.

**Comment by weshaggard** [comment]
I agree that for email addresses MailAddress is likely better although currently it throws on invalid email addresses. However this doc was intended as a RegEx example not a email parsing example. So if we would need to come up with another example if we were to switch. For better or worse parsing email addresses is a common RegEx example so I think we should just fix the current article. It might be worth adding a note that this is just an regex example but validating email addresses in .NET might be easier done with MailAddress.

**Comment by pkulikov** [comment]
By the way, the article already contains the note about `System.Net.Mail.MailAddress` and `MailAddress` constructor as an alternative to regex (see the note just above the [Compiling the code](https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format#compiling-the-code) section). Maybe it should be rephrased in such a way that makes the `MailAddress` using more preferable to the suggested regex example and placed at the start of the topic.

**Comment by pkulikov** [comment]
And a thought on high traffic, low satisfaction: most probably that comes from those folks who type in Google something like *.NET validate email address* and land in on this page, see that long-long regex, fight through the table to understand it, and then—at the end of the day—they see the note that kind of might make them angry of all the wasted time. Of course they would say that that page was not helpful. 

So, stating at the beginning what is the target audience and redirecting those Google "victims" might fix low satisfaction score.
